### PR TITLE
Added quickstart documentation & Tested QBE pipeline end-to-end with the new implementation of Aurum

### PR DESCRIPTION
**Pull request by snowgy**
- Addressed several bugs in the implementation of neighbor search functionality.
- Made necessary modifications to Kuzu statements by changing the node name from "Column" to "Dataset." This change was necessary as both "Column" and "Profile" are reserved terms in Kuzu.
- Included an initial quick start documentation based on zhiru's demo example, enabling users to run Ver end-to-end. Please note that View Distillation and View Presentation functionalities are not yet included in this version.

Caveat: While working with only four datasets related to Chicago public schools, Ver was able to identify an unexpectedly high number of 12,984 join graphs. Further investigation is needed to determine the reasons behind this abundance of join graphs.

Update: Previously, there is a bug in the join graph search implementation causing join graphs like [tbl1 join tbl2, tbl2 join tbl3, tbl3 join tbl1] (there is a circle in the graph). This bug is fixed right now, alleviating the problem of too many join graphs.

### CODE REVIEW

File: dindex_builder/graph_index_schema_kuzu.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
-CREATE NODE TABLE Column(id INT64, PRIMARY KEY (id));
+CREATE NODE TABLE Dataset(id INT64, PRIMARY KEY (id));
```
Comment (kevindharmawan) [inline dindex_builder/graph_index_schema_kuzu.txt:1]: I think using the name "Dataset" is a bit misleading, maybe we can use "DataColumn" or "DatasetColumn" or something else that can show that node represents column.

Comment (raulcf) [reply dindex_builder/graph_index_schema_kuzu.txt:1]: I like ColumnNode, it indicates a type of node in the graph, e.g., column.

File: archived/DoD/data_processing_utils.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 from collections import defaultdict
 import math
 
-from archived.DoD import FilterType
+# from archived.DoD import FilterType
```
Comment (raulcf) [inline archived/DoD/data_processing_utils.py:5]: if we don't need these we can clean them up directly (as opposed to comment them)

File: dindex_store/discovery_index.py [hunk 3]
```diff
@@ -122,6 +122,8 @@ def get_minhashes(self) -> Dict:
 
     def find_neighborhood(self, node_id: int, relation_type, hops: int = 1, desired_attributes=None):
         neighbors_ids = self.__graph_index.find_neighborhood(node_id, relation_type, hops)
+        if len(neighbors_ids) == 0:
+            return []
```
Comment (raulcf) [inline dindex_store/discovery_index.py:126]: I note that below (line 129) we return None while here we return an empty list. I'm not sure which one we want to return.

File: dindex_store/graph_index_kuzu.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 from typing import Dict, List
 from pathlib import Path
 import os
-
+import traceback
```
Comment (raulcf) [inline dindex_store/graph_index_kuzu.py:4]: maybe left from a previous debugging session?

File: quick_start_cli.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+## Setup
+Compatible with Python versions 3.8.10 and higher
+### Install dependencies
+```bash
+# Create an virtual environment 
+# (optional but recommended)
+$ python -m venv venv 
+$ source venv/bin/activate
+# install dependencies
+(venv) $ pip install -r requirements_quick_start.txt
```
Comment (raulcf) [inline quick_start_cli.md:11]: great, is this requirements_quick_start a cleaned-up version of the reqs?
Would it make sense to keep only one requirements file for the project?

Comment (raulcf) [reply quick_start_cli.md:11]: it would force us to start thinking about deployment too

File: dindex_store/discovery_index.py (hunk 3 above)
Comment (snowgy) [reply dindex_store/discovery_index.py:126]: ummm, I don't think we return None anywhere in this function. Line 129 fetches profiles for columns in `neighbors_ids`. line 125-126 is to deal with the corner case when the input node has no neighbors

File: quick_start_cli.md (hunk 5 above)
Comment (snowgy) [reply quick_start_cli.md:11]: requirements_quick_start is a cleaned-up version of the reqs for dindex_builder, dindex_store, qbe and ver cli. There would be more reqs after view-distillation and view-presentation are integrated to the quick start.

**Review by kevindharmawan — APPROVED** [review]

**Review by zaki-indra — APPROVED** [review]

**Pull request merged**